backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/PredicatesHandler.java

```diff
@@ -624,4 +637,79 @@ private String getEntityName() {
       return InternalFuture.failedStage(ex);
     }
   }
+
+  private InternalFuture<QueryFilterContext> setOwnerPredicate(long index, KeyValueQuery predicate)
+      throws ModelDBException {
+    var operator = predicate.getOperator();
+    InternalFuture<List<UserInfo>> userInfoListFuture;
+    if (operator.equals(OperatorEnum.Operator.CONTAIN)
+        || operator.equals(OperatorEnum.Operator.NOT_CONTAIN)) {
+      userInfoListFuture =
+          getFuzzyUserInfoList(predicate.getValue().getStringValue())
+              .thenApply(UserInfoPaginationDTO::getUserInfoList, executor);
+    } else {
+      var ownerIdsArrString = predicate.getValue().getStringValue();
+      List<String> ownerIds = new ArrayList<>();
+      if (operator.equals(OperatorEnum.Operator.IN)) {
+        ownerIds = Arrays.asList(ownerIdsArrString.split(","));
+      } else {
+        ownerIds.add(ownerIdsArrString);
+      }
+      userInfoListFuture =
+          getUserInfoFromAuthServer(
+                  new HashSet<>(ownerIds), Collections.emptySet(), Collections.emptyList())
+              .thenApply(userInfoMap -> new ArrayList<>(userInfoMap.values()), executor);
+    }
+
+    return userInfoListFuture.thenCompose(
+        userInfoList -> {
+          if (userInfoList != null && !userInfoList.isEmpty()) {
+            var resourceItemsFutures = new ArrayList<InternalFuture<Set<String>>>();
+            for (var userInfo : userInfoList) {
+              resourceItemsFutures.add(
+                  getResourceItemsForLoginUserWorkspace(
+                          userInfo.getVertaInfo().getUsername(),
```

Comment:
Not sure why we need this. `getResourceItemsForLoginUserWorkspace` will already give everything I have access to, right? I'm unsure why we need to query this for every user we find